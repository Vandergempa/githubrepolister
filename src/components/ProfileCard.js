import React from "react"

import Card from 'react-bootstrap/Card'
import Button from 'react-bootstrap/Button'

const ProfileCard = (props) => {
  const { userInfo } = props
  return (
    <Card className="col-xs-5 col-sm-6 m-5 text-center mb-auto">
      <Card.Img className="rounded-circle ml-auto mr-auto mt-2" variant="top" src={userInfo.avatar_url} style={{ width: '13rem' }} />
      <Card.Body>
        <Card.Title>{userInfo.name}</Card.Title>
        <Card.Text>
          <em>{userInfo.bio}</em>
        </Card.Text>
        <hr></hr>
        <Card.Text>
          Public repos: {userInfo.public_repos}
        </Card.Text>
        <Card.Text>
          Followers: {userInfo.followers} | Following: {userInfo.following}
        </Card.Text>
        <Card.Text>
          Created at: {userInfo.created_at}
        </Card.Text>
        <Button variant="primary" href={userInfo.html_url} >Go to profile</Button>
      </Card.Body>
    </Card>
  )
}

export default ProfileCard