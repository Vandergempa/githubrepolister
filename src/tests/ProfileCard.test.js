import React from 'react';
import { shallow } from 'enzyme';
import ProfileCard from '../components/ProfileCard';

let wrapper

beforeEach(() => {
  const userInfo = {
    "login": "octocat",
    "avatar_url": "https://github.com/images/error/octocat_happy.gif",
    "url": "https://api.github.com/users/octocat",
    "html_url": "https://github.com/octocat",
    "name": "monalisa octocat",
    "bio": "There once was...",
    "public_repos": 2,
    "followers": 20,
    "following": 0,
    "created_at": "2008-01-14T04:33:35Z",
    "updated_at": "2008-01-14T04:33:35Z"
  }

  wrapper = shallow(
    <ProfileCard
      userInfo={userInfo}
    />
  );
});

test('Should render ProfileCard component correctly', () => {
  expect(wrapper).toMatchSnapshot();
});