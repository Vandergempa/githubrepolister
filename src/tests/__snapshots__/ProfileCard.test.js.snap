// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ProfileCard component correctly 1`] = `
<Card
  body={false}
  className="col-xs-5 col-sm-6 m-5 text-center mb-auto"
>
  <CardImg
    className="rounded-circle ml-auto mr-auto mt-2"
    src="https://github.com/images/error/octocat_happy.gif"
    style={
      Object {
        "width": "13rem",
      }
    }
    variant="top"
  />
  <CardBody>
    <CardTitle>
      monalisa octocat
    </CardTitle>
    <CardText>
      <em>
        There once was...
      </em>
    </CardText>
    <hr />
    <CardText>
      Public repos: 
      2
    </CardText>
    <CardText>
      Followers: 
      20
       | Following: 
      0
    </CardText>
    <CardText>
      Created at: 
      2008-01-14T04:33:35Z
    </CardText>
    <Button
      active={false}
      disabled={false}
      href="https://github.com/octocat"
      type="button"
      variant="primary"
    >
      Go to profile
    </Button>
  </CardBody>
</Card>
`;
